test/library/packages/ProtobufProtocolSupport/endToEnd/typesTest/read.chpl

```diff
@@ -1,7 +1,7 @@
 use typesTest;
 use IO;
 
-var messageObj = new Types();
+var messageObj = new typesTest.Types();
```

Comment:
yes

Reply:
I'm not following which change led to this change of behavior...  Oh wait, is it because `ChapelStandard` contains `public use`s of all its modules rather than simply `use`s, so now they're equally close via ChapelStandard's implicit `use` than typesTest's, which is private?  If so, if we changed the public uses to `public use ModuleName as _;` would that be another way of resolving this while also making the auto-used module names not be so close to user code?  (without preventing them from being available from the program scope)?   Or if we made `public use` not inject the module name as I've been musing about on chat with you?

Reply:
(also, where is the typesTest module defined?  I wasn't able to find it with some quick searches)

Reply:
I believe the typesTest module is created by Protobuf for this test, I think based on one of the other files in this hierarchy (but I was having trouble finding it as well).

> If so, if we changed the public uses to `public use ModuleName as _;` would that be another way of resolving this while also making the auto-used module names not be so close to user code?

I think that would work, but it would make it harder for a user to use qualified access to choose an automatically provided symbol over another one with the same name.  They'd have to insert an explicit use or import of the module to do so

Reply:
> I'm not following which change led to this change of behavior...  Oh wait, is it because `ChapelStandard` contains `public use`s of all its modules rather than simply `use`s, so now they're equally close via ChapelStandard's implicit `use` than typesTest's, which is private?

Yes

> If so, if we changed the public uses to `public use ModuleName as _;` would that be another way of resolving this while also making the auto-used module names not be so close to user code?  (without preventing them from being available from the program scope)?   Or if we made `public use` not inject the module name as I've been musing about on chat with you?

I would assume that if `Types` is an auto-used standard module that `Types.bla` should work. As a result that symbol `Types` would still conflict here. (This conversation strikes me as a bit related to #19313).

The `testTypes` module is generated by the protobuf compiler in this particular test case.